### ISSUE TITLE
LH crashes on non-ascii string literals

### ISSUE THREAD

**Issue by gridaphobe** [ucsd-progsys/liquidhaskell#813]
see http://goto.ucsd.edu:8090/index.html#?demo=permalink%2F1470924649_14708.hs

**Comment by nikivazou** [comment]
I run into this too. Is there any progress or should I take a look?

**Comment by jekor** [comment]
This might be unrelated to the fix in #826, but I get the following error with non-ASCII string literals on 0.9.6.3:

```
<no location info>: error:
    src/Infer/.liquid/Value.hs.cst.html: withFile: invalid argument (cannot decode byte sequence starting from 194)
```

This error does not appear without `liquid-prelude`. Characters tested: `±`, `天`.
